appscale/tools/appscale.py

```diff
@@ -764,12 +757,19 @@ def destroy(self):
     if "EC2_URL" in contents_as_yaml:
       os.environ["EC2_URL"] = contents_as_yaml["EC2_URL"]
 
-    if 'keyname' in contents_as_yaml:
-      command.append("--keyname")
-      command.append(contents_as_yaml['keyname'])
+    if clean:
+      LocalState.ensure_user_wants_to_clean()
+      all_ips = LocalState.get_all_public_ips(keyname)
+      for ip in all_ips:
+        RemoteHelper.ssh(ip, keyname, self.TERMINATE, is_verbose)
+      AppScaleLogger.success("Successfully cleaned your AppScale deployment.")
 
-    if 'verbose' in contents_as_yaml and contents_as_yaml['verbose'] == True:
-      command.append("--verbose")
+    if terminate:
+      infrastructure = LocalState.get_infrastructure(keyname)
+      if infrastructure != "xen" and not LocalState.are_disks_used(
```

Comment:
Yes, the check is to ensure that if we terminate an ephemeral instance (terminate == true), the user is aware that data can be lost if no volume has been used.